engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -448,6 +450,14 @@ public void startApplication() throws YarnException, IOException
       }
       String libJarsCsv = copyFromLocal(fs, appPath, localJarFiles.toArray(new String[]{}));
 
+      if (UserGroupInformation.isSecurityEnabled() && customKeystoreConfig != null && customKeystorePath != null) {
+        String[] customSSLConfArr = {customKeystoreConfig, customKeystorePath};
+        String customSSLFileConf = copyFromLocal(fs, appPath, customSSLConfArr);
+        String configFileName = (new java.io.File(customKeystoreConfig)).getName();
+        dag.getAttributes().put(Context.DAGContext.STRAM_HTTP_CUSTOM_CONFIG, configFileName);
```

Comment:
"If you think that naming in the API and user facing elements that are subject to backward compatibility isn't that important"

I am not saying that. In fact, I am arguing the opposite and feel you are following the same behavior that you are warning against. You are looking at this attribute by itself and its future impact but not considering the impact on existing users and how it can be mitigated. There is already a naming convention the existing users are used to with the web service properties with the authentication attribute and this is not the only attribute. For example, I didn't see any suggestion from you on how to mitigate the impact, like deprecating the old authentication attribute in favor of a new one with the new naming convention.

Like I mentioned earlier, I am not against having a different name, but this can't be done in isolation and would need to be done along with the changes I described above (can be in separate PRs but within the release). I am not in favor of just HTTP_ as it doesn't capture the functionality properly and prefer something like MANAGEMENT_ or WEBSERVICE_. STRAM_HTTP_AUTHENTICATION could map to MANAGEMENT_AUTHENTICATION and STRAM_HTTP_CUSTOM_CONFIG could map to MANAGEMENT_HTTP_CUSTOM_CONFIG as it deals with http protocol specific configuration.